build.gradle

```diff
@@ -324,3 +328,78 @@ tasks.withType(Test) {
         }
     }
 }
+
+
+task checkForMissingRunConfigurations() {
+    def runConfigurationsDir = new File(".idea/runConfigurations")
+    if (!runConfigurationsDir.exists()) {
+        generateRunConfigurations()
+    }
+}
+
+task regenerateRunConfigurations() {
+    doLast {
+        generateRunConfigurations()
+    }
+}
+
+private void generateRunConfigurations() {
+
+
+
+    createGradleRunConfiguration('Run IDE (default)', 'runIde', '', '')
+    createGradleRunConfiguration('Run Tests (default)', 'clean test', '', '')
+
+    createGradleRunConfiguration('Generate Parsers', 'cleanGenerated generateHaxeParser generateHxmlParser', '', '')
+
+
+    createGradleRunConfiguration('Run Tests 2018', 'clean test', "-PtargetVersion=$latest2018Version", '')
+    createGradleRunConfiguration('Run Tests 2017', 'clean test', "-PtargetVersion=$latest2017Version", '')
+    createGradleRunConfiguration('Run Tests 2016', 'clean test', "-PtargetVersion=$latest2016Version", '')
+
+    createGradleRunConfiguration('Run IDE 2018',
+                                 'runIde',
+                                 "-PtargetVersion=$latest2018Version",
+                                 ' -DHAXELIB_LIST_PATH_SUPPORTED=1')
+    createGradleRunConfiguration('Run IDE 2017',
+                                 'runIde',
+                                 "-PtargetVersion=$latest2017Version",
+                                 '-DHAXELIB_LIST_PATH_SUPPORTED=1')
+    createGradleRunConfiguration('Run IDE 2016',
+                                 'runIde',
+                                 "-PtargetVersion=$latest2016Version",
+                                 '-Xmx1024m -Xms512m -XX:MaxPermSize=256m -ea -DHAXELIB_LIST_PATH_SUPPORTED=1')
```

Comment:
For these, it would be really nice to add `-Didea.is.internal=true` which turns on the PSI viewer in the Tools menu.